solarforecastarbiter/reference_forecasts/models.py

```diff
@@ -59,16 +59,34 @@
 and that functions that accept primitives may be easier to maintain in
 the long run.
 """
+from functools import partial, wraps
+import inspect
 
-from functools import partial
 
 from solarforecastarbiter import pvmodel
 from solarforecastarbiter.io.nwp import load_forecast
 from solarforecastarbiter.reference_forecasts import forecast
 
+
 import pandas as pd
 
 
+def set_nwp_model(nwp_model):
+    def dec(f):
+        @wraps(f)
+        def wrapper(*args, **kwargs):
+            return f(*args, **kwargs)
+        wrapper.__nwp_model__ = nwp_model
+        return wrapper
+    return dec
+
+
+def get_nwp_model(func=None):
+    if func is None:
+        func = globals()[inspect.stack()[1].function]
```

Comment:
ok, now there is a kwarg only parameter '__model' that defines the model_str. I like keeping the model str on/in the function instead of in a separate dictionary